### ISSUE TITLE
`tendermint-testgen`: add support for `no_std`

### ISSUE THREAD

**Issue by plafer** [informalsystems/tendermint-rs#1258]
### Description
It would be great if the `tendermint-testgen` crate supported `no_std`. This is currently a [blocker in ibc-rs](https://github.com/cosmos/ibc-rs/pull/318#issuecomment-1373073729) for making our `mocks` feature `no_std`.

### Definition of "done"

```toml
tendermint-testgen = {version = "0.x", default-features = false }
```
doesn't link `std`.

**Comment by thanethomson** [comment]
@plafer do you perhaps have capacity to implement this?

**Comment by plafer** [comment]
Yes, happy to do it whenever I find the time. However, note that this is not high priority in ibc-rs right now.
